Feature/comment reaction action

tulga-bytes opened the pull request:
Add AI-powered PR Assistant GitHub Action that automatically implements code changes based on PR comments using the Auggie SDK.

## Key Features

- **AI-Powered Implementation**: Integrates Auggie SDK to understand and implement requested changes from PR comments
- **Context-Aware Processing**: Gathers full PR context including diff, changed files, and comment threads
- **Automatic Git Operations**: Commits and pushes changes directly to the PR branch with descriptive commit messages
- **Visual Feedback**: Adds emoji reactions to comments for immediate acknowledgment
- **Status Notifications**: Posts success or failure comments to the PR with detailed information
- **Full TypeScript Implementation**: Type-safe code with comprehensive error handling

## Technical Implementation

- Composite GitHub Action with Node.js 22 runtime
- Supports both `pull_request_review_comment` and `issue_comment` events
- Configurable reaction types (👀, 🚀, ❤️, 🎉, etc.)
- Streams Auggie tool execution for detailed logging
- Includes complete documentation and integration guide

## Usage Example

```yaml
- name: PR Assistant
  uses: augmentcode/augment-agent/assistant@feature/comment-reaction-action
  with:
    github_token: ${{ secrets.GITHUB_TOKEN }}
    comment_id: ${{ github.event.comment.id }}
    event_name: ${{ github.event_name }}
    augment_api_token: ${{ secrets.AUGMENT_API_KEY }}
    augment_api_url: ${{ secrets.AUGMENT_API_URL }}
```

Users can trigger the assistant by commenting `@augment <instruction>` on a PR, and the action will automatically implement the requested changes.

---
*🤖 This description was generated automatically. Please react with 👍 if it's helpful or 👎 if it needs improvement.*

tulga-bytes committed:
feat: add comment-reaction action for responding to user comments

- Create new composite action for adding emoji reactions to GitHub comments
- Support both PR review comments and issue comments
- Configurable reaction type with sensible default (eyes)
- Full TypeScript implementation with type safety
- Comprehensive error handling and logging
- Includes README with usage examples and documentation

tulga-bytes committed:
docs: add integration guide for comment-reaction action

tulga-bytes committed:
refactor: rename comment-reaction to pr-assistant

- Rename directory from comment-reaction to pr-assistant
- Update parseInt to Number.parseInt for consistency
- Prepare for expanded functionality to implement PR changes based on comments

tulga-bytes committed:
refactor: rename pr-assistant to assistant

Simplify naming from pr-assistant to assistant

tulga-bytes committed:
feat: add Augment API credentials as inputs

- Add augment_api_token and augment_api_url as optional inputs
- Forward credentials as environment variables to the action
- Enable Augment authentication for future functionality

tulga-bytes committed:
feat: enhance assistant to implement PR changes using Auggie SDK

Major enhancements:
- Add Auggie SDK integration to implement changes based on comments
- Gather comprehensive PR context (diff, files, comments, metadata)
- Automatically commit and push changes to PR branch
- Add success/failure comments to PR
- Configure git with GitHub Actions bot credentials
- Handle both issue_comment and pull_request_review_comment events

Dependencies:
- Add @augmentcode/auggie-sdk for AI-powered code changes
- Add @actions/exec for git operations

The action now:
1. Adds eyes reaction to show processing
2. Gathers PR context and comment thread
3. Invokes Auggie with full context
4. Commits and pushes implemented changes
5. Posts result comment to PR

tulga-bytes committed:
docs: update README for PR Assistant functionality

- Update description to reflect AI-powered implementation
- Add comprehensive usage examples
- Document how the action works step-by-step
- Add setup instructions and troubleshooting guide
- Include example instructions for common use cases

tulga-bytes committed:
fix: add reaction immediately for quick user feedback

Move the reaction step to happen right after Octokit initialization,
before any other processing or API calls. This ensures users get
immediate visual feedback that their comment is being processed.

tulga-bytes committed:
refactor: move reaction to separate action.yml step for instant feedback

Move the emoji reaction from TypeScript code to a separate github-script
step in action.yml. This provides even faster user feedback since:
- No need to wait for Node.js setup
- No need to wait for npm install
- Reaction happens immediately before any processing

Benefits:
- Reaction appears in ~1-2 seconds instead of ~10-15 seconds
- Cleaner separation of concerns
- Simpler TypeScript code (removed unused reaction logic)
- Better user experience with instant visual feedback

tulga-bytes committed:
fix: install Auggie CLI before running the action

Add step to install @augmentcode/auggie globally before running the
assistant. The Auggie SDK spawns the auggie CLI binary, which needs
to be installed on the runner.

This fixes the 'spawn auggie ENOENT' error.

tulga-bytes committed:
fix: set isAnswerOnly to false to actually implement changes

Change isAnswerOnly from true to false in auggie.prompt() call.
When isAnswerOnly is true, Auggie only provides a text description
of what it would do without actually making file changes.

With isAnswerOnly: false, Auggie will:
- Actually edit the files
- Make the requested changes
- Allow git to detect and commit the changes

tulga-bytes committed:
fix: improve instruction to explicitly require file changes

After reviewing the Auggie SDK source code, isAnswerOnly only controls
which text is returned, NOT whether tools are executed. The real issue
was that the instruction wasn't explicit enough.

Changes:
- Add clear context that this is an AI assistant implementing changes
- Explicitly state 'You MUST actually implement the requested changes'
- Tell Auggie to use file editing tools (str-replace-editor, save-file)
- Clarify NOT to just describe what would be done
- Set isAnswerOnly back to true (correct setting for clean final answer)

This ensures Auggie understands it needs to actually edit files, not
just provide a description of what it would do.

tulga-bytes committed:
feat: add streaming logs for tool usage

Add onSessionUpdate callback to stream tool execution in real-time.
This provides visibility into what Auggie is doing during execution.

Logs include:
- 🔧 When a tool starts executing (with input parameters)
- ✅ When a tool completes (with output, truncated if long)
- Real-time progress updates during long-running operations

Benefits:
- Users can see what's happening in the workflow logs
- Easier to debug when things go wrong
- Better transparency into Auggie's actions
- Helps understand which tools are being used

Inspired by examples/dev.ts in the Auggie SDK.

tulga-bytes committed:
fix: add delay before closing Auggie to ensure file writes complete

The Auggie SDK uses SIGKILL to terminate the process immediately,
which may not give enough time for file writes to be flushed to disk.

Changes:
- Log the workspace root directory for debugging
- Add 2-second delay after prompt completes before closing
- This ensures all file operations have time to complete
- Prevents 'no changes to commit' issue

The delay allows the Auggie process to finish writing all file
changes to disk before being terminated.

tulga-bytes committed:
fix: correct change detection logic in commitAndPush function

- Replace unreliable git diff --cached --quiet check with git status --porcelain
- Add better logging to show detected changes
- Fixes issue where changes were made but not committed due to incorrect detection

tulga-bytes committed:
fix: pull with rebase before pushing to handle remote changes

- Add git pull --rebase before push to sync with remote branch
- Prevents non-fast-forward errors when remote has new commits
- Gracefully handles rebase failures and attempts push anyway

tulga-bytes committed:
refactor: remove rebase logic, rely on workflow checkout instead

- Remove git pull --rebase step from commitAndPush
- Workflow now ensures latest PR state is checked out
- Simpler and more reliable approach

tulga-bytes committed:
feat: quote original comment in success/failure replies

- Format original comment as quoted text in reply
- Makes it clear which request was fulfilled or failed
- Improves conversation flow in PR comments

tulga-bytes committed:
style: fix code formatting issues

- Run prettier to fix formatting in assistant files
- Fixes format-check CI failure

justinxu421 commented on assistant/action.yml:
It's probably fine for this to live here for now, but to host it on marketplace, I believe the 1 click install will only link to the mainline action https://github.com/marketplace/actions/augment-agent i.e. `augmentcode/augment-agent@v0.1.3`, so this nested one might not be as discoverable.

<img width="674" height="307" alt="Image" src="https://github.com/user-attachments/assets/fe4d47c2-a4d9-42be-b5ff-5e74fdad927f" />

We've opted to maintain several repos and thus publish separate marketplaces to get around this the code review action:
- https://github.com/marketplace/actions/augment-review-pr
- https://github.com/augmentcode/review-pr/blob/main/action.yml

But it's also more work to maintain all the repos and version them. Doing it this way could also be a way to host it without fully publicizing it, so it could be a good middleground.

Just food for thought.

justinxu421 commented on assistant/action.yml:
```diff
@@ -0,0 +1,92 @@
+    description: "The reaction type to add (e.g., eyes, rocket, heart, hooray, confused, +1, -1, laugh)"
+    required: false
+    default: "eyes"
+  augment_api_token:
+    description: "API token for Augment services. Store as repository secret for security."
+    required: false
+  augment_api_url:
+    description: "URL endpoint for Augment API requests. Store as repository variable."
```
Do we want to add support for rules/mcp? Or just keep it simple for now

justinxu421 commented on assistant/src/index.ts:
```diff
@@ -0,0 +1,494 @@
+
+      // Step 6: Commit and push changes
+      await commitAndPush(commentId, context.headBranch, githubToken, owner, repo);
+
```
I wonder if directly committing to the repo is potentially too disruptive to some workflows and if we should make this configurable as an option.

Another option is to create a PR on top of the existing one and allow the user to merge it in if it's useful?

justinxu421 replied:
This kind of thing came up in a discussion with @akshayutture-augment on suggested auggie fixes

justinxu421 commented on assistant/src/index.ts:
```diff
@@ -0,0 +1,494 @@
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import { Octokit } from '@octokit/rest';
+import { Auggie } from '@augmentcode/auggie-sdk';
```
Being able to use the auggie-sdk here is pretty neat

justinxu421 commented on assistant/INTEGRATION_GUIDE.md:
I don't think we need this / agent generated?

tulga-bytes replied:
Yeah this was agent generated gonna remove.

justinxu421 commented on assistant/src/index.ts:
should we add test so some of these functions? (we probably also need to add some CI to this repo if so)

tulga-bytes replied:
I'm gonna move things to new repository and address this in that repo.

https://github.com/augmentcode/augment-action repository

justinxu421 commented:
auggie review

justinxu421 commented on assistant/src/index.ts:
```diff
@@ -0,0 +1,506 @@
+
+  // Push changes
+  core.info(`🚀 Pushing to ${headBranch}...`);
+  const remoteUrl = `https://x-access-token:${githubToken}@github.com/${owner}/${repo}.git`;
```
probably fine, but probably an easy change to implement here

tulga-bytes committed:
feat: add support for /pr command to create PR targeting original PR

- Add comment_body input parameter to action.yml
- Implement createPRForChanges function to create new branch and PR
- Update main function to detect /commit vs /pr command
- For /pr: create new branch, commit changes, and create PR targeting original PR branch
- For /commit: maintain existing behavior of committing directly to PR branch

tulga-bytes committed:
refactor: let Auggie agent create PR instead of action code

- Remove createPRForChanges function
- Update invokeAuggie to accept shouldCreatePR parameter
- For /pr command, instruct Auggie to create branch, commit, and create PR using its tools
- This avoids needing PAT_TOKEN since Auggie has GitHub API access
- Simplifies setup and makes it work out of the box

augmentmoogi commented on assistant/README.md:
````diff
@@ -0,0 +1,190 @@
+
+```yaml
+- name: PR Assistant
+  uses: augmentcode/augment-agent/assistant@feature/comment-reaction-action
````
This doesn't feel right to me.

tulga-bytes replied:
I decided to create new repo for the comment actions and going to move things there.

https://github.com/augmentcode/augment-action

augmentmoogi commented on assistant/src/index.ts:
```diff
@@ -0,0 +1,525 @@
+#!/usr/bin/env node
```
I recommend to split this

1. Put everything non-augment away in a different file
2. Everything augment keep here

It's hard to see the actual integration here.

tulga-bytes replied:
Agreed. I'm gonna re-organize things

augmentmoogi commented on assistant/src/index.ts:
```diff
@@ -0,0 +1,525 @@
+  }
+
+  // Build context for Auggie
+  const baseInstruction = `You are an AI assistant helping to implement code changes in a Pull Request.
```
I recommend to pull this out to a separate file for readability.

tulga-bytes replied:
Will address it as part of re-organizing
